packages/now-node/src/launcher.ts

```diff
@@ -8,108 +12,148 @@ type LauncherConfiguration = {
+        Server.prototype.listen = originalListen;
+        const server = listener;
+        bridge.setServer(server);
+        bridge.listen();
+      } else if (typeof listener === 'function') {
+        Server.prototype.listen = originalListen;
+        let server: Server;
+        if (shouldAddHelpers) {
+          bridge = new Bridge(undefined, true);
+          server = require(helpersPath).createServerWithHelpers(
+            listener,
+            bridge
+          );
+        } else {
+          server = createServer(listener);
+        }
+        bridge.setServer(server);
+        bridge.listen();
+      } else if (
+        typeof listener === 'object' &&
+        Object.keys(listener).length === 0
+      ) {
+        setTimeout(() => {
+          if (!isServerListening) {
+            console.error('No exports found in module %j.', entrypointPath);
+            console.error('Did you forget to export a function or a server?');
+            process.exit(1);
+          }
+        }, 5000);
+      } else {
+        console.error('Invalid export found in module %j.', entrypointPath);
+        console.error('The default export must be a function or server.');
       }
-    }, 5000);
-  } else {
-    console.error('Invalid export found in module "${entrypointPath}".');
-    console.error('The default export must be a function or server.');
-  }
-} catch (err) {
-  if (err.code === 'MODULE_NOT_FOUND') {
-    console.error(err.message);
-    console.error('Did you forget to add it to "dependencies" in \`package.json\`?');
-    process.exit(1);
-  } else {
-    console.error(err);
-    process.exit(1);
-  }
+    } catch (err) {
+      if (err.code === 'MODULE_NOT_FOUND') {
+        console.error(err.message);
+        console.error(
+          'Did you forget to add it to "dependencies" in `package.json`?'
+        );
+      } else {
+        console.error(err);
+      }
+      process.exit(1);
+    }
+
+    return bridge;
+  };
 }
 
-exports.launcher = bridge.launcher;`;
+export function makeAwsLauncher(config: LauncherConfiguration): string {
+  const { entrypointPath, awsLambdaHandler = '' } = config;
+  return `const url_1 = require("url");
+const funcName = ${JSON.stringify(awsLambdaHandler.split('.').pop())};
+const entrypointPath = ${JSON.stringify(entrypointPath)};
+exports.launcher = ${getAwsLauncher(config)}`;
 }
 
-export function makeAwsLauncher({
+export function getAwsLauncher({
   entrypointPath,
   awsLambdaHandler = '',
-}: LauncherConfiguration): string {
+}: LauncherConfiguration) {
   const funcName = awsLambdaHandler.split('.').pop();
-  return `const url = require("url");
-  exports.launcher = function (e, context, callback) {
-    const zeitNowEvent = JSON.parse(e.body);
-    const { path, method: httpMethod, body, headers } = zeitNowEvent;
-    const { query } = url.parse(path, true);
-    const queryStringParameters = {};
+  if (typeof funcName !== 'string') {
+    throw new TypeError('Expected "string"');
```

Comment:
Was this added because typescript was complaining? I can't think of a scenario where `awsLambdaHandler` is a string and then `funcName` is not a string 🤔

Reply:
Ya it's to make TypeScript happy.

Reply:
I guess it's because `.pop()` _could_ be undefined, and TS isn't smart enough to know that it'll always be a string in this case.